pkg/distributor/ha_tracker_test.go

```diff
@@ -35,19 +40,22 @@ outer:
 			r := c.elected[key]
 			c.electedLock.RUnlock()
 
+			// If the replica or the timestamp don't match, we save the error
+			// to return if the expected match is not found within the timeout period
 			if r.GetReplica() != replica {
 				err = fmt.Errorf("replicas did not match: %s != %s", r.GetReplica(), replica)
 				continue outer
-			}
-			if timestamp.Time(r.GetReceivedAt()).Equal(expected) {
+			} else if !timestamp.Time(r.GetReceivedAt()).Equal(expected) {
```

Comment:
Tiny detail, but using 'else if' when previous 'if' body does continue, break or return is not very nice. Previous version was clearer.

Reply:
Right! Fixed